src/Altinn.App.Core/Helpers/ObjectUtils.cs

```diff
@@ -10,13 +10,25 @@ public static class ObjectUtils
     /// <summary>
     /// Recursively initialize all <see cref="List{T}"/> properties on the object that are currently null
     /// Also ensure that all string properties that are empty are set to null
+    /// And set empty Guid properties named "AltinnRowId" to a new random guid
     /// </summary>
     /// <param name="model">The object to mutate</param>
-    public static void InitializeListsAndNullEmptyStrings(object model)
+    public static void InitializeAltinnRowId(object model)
     {
+        ArgumentNullException.ThrowIfNull(model);
+
         foreach (var prop in model.GetType().GetProperties())
         {
-            if (prop.PropertyType.IsGenericType && prop.PropertyType.GetGenericTypeDefinition() == typeof(List<>))
+            if (prop.PropertyType == typeof(Guid) && prop.Name == "AltinnRowId")
```

Comment:
Perhaps a simple refactor of this to a `PropertyIsAltinnRowGuid(PropertyInfo prop)` just remove the duplicated logic in the `RemoveAltinnRowIds `method and keep the string name in one place.

```
    private static bool PropertyIsAltinnRowGuid(System.Reflection.PropertyInfo prop)
    {
        return prop.PropertyType == typeof(Guid) && prop.Name == "AltinnRowId";
    }
```

Reply:
Done